modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ConcurrentLaws.kt

```diff
@@ -674,4 +659,41 @@ object ConcurrentLaws {
         )).parSequence(ListK.traverse()).unit()
       }.equalUnderTheLaw(unit(), EQ)
     }
+
+  fun <F> Concurrent<F>.onErrorIsRunWhenErrorIsRaised(EQ: Eq<Kind<F, Int>>, ctx: CoroutineContext) =
+    forAll(Gen.int()) { i ->
+      fx.concurrent {
+
+        val startLatch = Promise<F, Unit>(this@onErrorIsRunWhenErrorIsRaised).bind()
+        val errorLatch = Promise<F, Int>(this@onErrorIsRunWhenErrorIsRaised).bind()
+
+        startLatch.complete(Unit).flatMap { raiseError<Exception>(RuntimeException("Boom")) }
+          .onError(errorLatch.complete(i))
+          .fork(ctx).bind()
+
+        startLatch.get().bind() // Waits on promise of `use`
+
+        val waitExit = errorLatch.get().bind()
+        waitExit
+      }.equalUnderTheLaw(just(i), EQ)
+    }
+
+  fun <F> Concurrent<F>.onErrorIsNotRunByDefault(EQ: Eq<Kind<F, Tuple2<Int, Boolean>>>, ctx: CoroutineContext) =
+    forAll(Gen.int()) { i ->
+
+      val CF = this@onErrorIsNotRunByDefault
+      fx.concurrent {
+
+        val startLatch = Promise<F, Int>(CF).bind()
+        val onErrorRun = Ref(false).bind()
+
+        val (completed, _) = startLatch.complete(i)
+          .onError(onErrorRun.set(true))
+          .fork(ctx).bind()
+
+        completed.bind()
+
+        startLatch.get().bind() toT onErrorRun.get().bind()
+      }.equalUnderTheLaw(just(i toT false), EQ)
+    }
```

Comment:
Could you please add an additional law to check that

```
raiseError(e).flatMap { raiseError(e).onError(...) }.onError(...)
```

Triggers both. Maybe with an atomic counter.